talisker/endpoints.py

```diff
@@ -247,7 +260,222 @@ def metrics(self, request):
+                ),
+            ))
+
+        return html_response(
+            'Python Packages',
+            table(
+                rows,
+                headers=['Package', 'Version', 'Location', 'PyPI Link'],
+            )
+        )
+
+    @private
+    def workers(self, request):
+        """Information about workers resource usage."""
+        import psutil
+        arbiter = psutil.Process(os.getppid())
+        workers = arbiter.children()
+        workers.sort(key=lambda p: p.pid)
+
+        rows = [format_psutil_row('Gunicorn Master', arbiter)]
+        for i, worker in enumerate(workers):
+            rows.append(format_psutil_row('Worker {}'.format(i), worker))
+
+        master = arbiter.as_dict(MASTER_FIELDS)
+        master['cmdline'] = ' '.join(master['cmdline'])
+
+        environ = master.pop('environ')
+        if 'SENTRY_DSN' in environ:
+            environ['SENTRY_DSN'] = sanitize_url(environ['SENTRY_DSN'])
+        clean_environ = [
+            (k, v) for k, v in sorted(environ.items())
+            if k in TALISKER_ENV_VARS
+        ]
+        sorted_master = [(k, master[k]) for k in MASTER_FIELDS if k in master]
+
+        return html_response(
+            'Workers',
+            ['<h2>Workers</h2>'],
+            table(rows, headers=HEADERS),
+            ['<h2>Process Information</h2>'],
+            table(sorted_master),
+            ['<h2>Process Environment (whitelist)</h2>'],
+            table(clean_environ),
+        )
+
+    @private
+    def objgraph(self, request):
+        import objgraph
+        limit = int(request.args.get('limit', 10))
+        types = objgraph.most_common_types(limit=limit, shortnames=False)
+        leaking = objgraph.most_common_types(
+            limit=limit,
+            objects=objgraph.get_leaking_objects(),
+            shortnames=False,
+        )
+
+        limits = '<p>Number of items: {}, {}, {}</p>'.format(
+            link('{}', request.path + '?limit={}', 10),
+            link('{}', request.path + '?limit={}', 20),
+            link('{}', request.path + '?limit={}', 50),
+        )
+        return html_response(
+            'Python Objects',
+            ['<h1>Python Objects for Worker pid {}</h1>'.format(os.getpid())],
+            [limits],
+            ['<h2>Most Common Objects</h2>'],
+            table(types),
+            ['<h2>Leaking Objects (no referrer)</h2>'],
+            table(leaking),
+        )
+
+
+# diy html templating
+CDN = '//cdnjs.cloudflare.com/ajax/libs'
+HEADER = """
+<head>
+    <title>Talisker: {title}</title>
+    <link rel="stylesheet"
+    href="{cdn}/twitter-bootstrap/4.1.1/css/bootstrap.min.css">
+    <link rel="stylesheet"
+    href="{cdn}/bootstrap-table/1.12.1/bootstrap-table.min.css">
+    <script
+    src="{cdn}/jquery/3.3.1/jquery.js"></script>
+    <script
+    src="{cdn}/twitter-bootstrap/4.1.1/js/bootstrap.min.js"></script>
+    <script
+    src="{cdn}/bootstrap-table/1.12.1/bootstrap-table.min.js"></script>
+</head>
```

Comment:
eek, that's a lot of dependencies

Reply:
I'd love a css only pretty table, but I couldn't find one.